metrics/providers/micrometer/src/main/java/io/helidon/metrics/providers/micrometer/MMeterRegistry.java

```diff
@@ -512,10 +552,47 @@ void onMeterRemoved(Meter removedMeter) {
                 recordRemove(removedHelidonMeter);
             }
         } finally {
-            lock.unlock();
+            lock.writeLock().unlock();
         }
     }
 
+    private io.helidon.metrics.api.Meter noopMeterIfDisabled(io.helidon.metrics.api.Meter.Builder<?, ?> builder) {
+        if (!isMeterEnabled(builder.name(), builder.tags(), builder.scope())) {
+
+            io.helidon.metrics.api.Meter result = metricsFactory.noOpMeter(builder);
+            try {
+                onAddListeners.forEach(listener -> listener.accept(result));
+            } catch (Exception e) {
+                // Ignored.
```

Comment:
Ding ding ding. My alarm bells go off; worthy of a log at least I would think

Reply:
[I responded to this comment earlier but maybe I didn't save it.]

I've removed the try/catch block. Everywhere else where we invoke the listeners we (intentionally) let any exceptions bubble up. We should do the same here.